main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -40,10 +40,14 @@ object Simplifier {
     /**
       * Translates the given definition `def0` to the SimplifiedAst.
       */
-    def visitDef(def0: LoweredAst.Def): SimplifiedAst.Def = {
-      val fs = def0.spec.fparams.map(visitFormalParam)
-      val exp = visitExp(def0.impl.exp)
-      SimplifiedAst.Def(def0.spec.ann, def0.spec.mod, def0.sym, fs, exp, def0.impl.inferredScheme.base, def0.sym.loc)
+    def visitDef(defn: LoweredAst.Def): SimplifiedAst.Def = defn match {
+      case LoweredAst.Def(sym, spec, impl) =>
+        val fs = spec.fparams.map(visitFormalParam)
+        val exp = visitExp(impl.exp)
+        val funType = impl.inferredScheme.base
+        val retType = funType.arrowResultType
```

Comment:
Remind me why we cannot get this from the spec. I thought spec and impl have the same schemes after monomorph?

Reply:
to maintain current types, this can be changed later